command/agent/command.go

```diff
@@ -1165,6 +1164,22 @@ func (c *Command) setupTelemetry(config *Config) (*metrics.InmemSink, error) {
 		telConfig = config.Telemetry
 	}
 
+	// The values are defaulted in both the dev and default config object, so
+	// we should never have empty values. Do this, so it can never happen and
+	// cause agents to panic on start.
+	inMemInt := 10 * time.Second
+	inMemRet := 1 * time.Minute
```

Comment:
I get that we're trying to be belt-and-suspenders here but this leaves two different places where the default value is being defined. Is there a reason we expect the default/dev config objects not to be enough?

Reply:
In my opinion no, the default/dev config objects should be enough as we always overlay user passed config options on a default config. The reason I put this in is because the telemetry setup also includes code to handle a nil telemetry config object, which made me suspicious of why it was added. I am happy to remove this, the nil check is potentially a remanent of code added ~10 years ago and not touched since.

Reply:
Yeah, I can't think of a way we could end up with a `nil` here. Even with an explicitly empty block the default config is merged into it.

Any misconfiguration at this point would also fail loudly and early right? Meaning the agent wouldn't start and no state would be modified, so even if an error does happen it seems like it would fail safely.

If we do want to be extra safe here we could instead have a `DefaultTelemetry()` function that can be called in both `DefaultConfig` and inside the `if` above.